### PR TITLE
dbg macro: Discuss use in tests, and slightly clarify

### PR DESCRIPTION
**Pull request by ijackson**
As discussed in a tangent in #82778.

I chose to use [semantic newlines](https://rhodesmill.org/brandon/2012/one-sentence-per-line/) in the source text but I don't mind reformatting it.

### CODE REVIEW

File: library/std/src/macros.rs [hunk 1]
```diff
@@ -185,9 +185,10 @@ macro_rules! eprintln {
 /// builds or when debugging in release mode is significantly faster.
 ///
 /// Note that the macro is intended as a debugging tool and therefore you
-/// should avoid having uses of it in version control for long periods.
-/// Use cases involving debug output that should be added to version control
-/// are better served by macros such as [`debug!`] from the [`log`] crate.
+/// should avoid having uses of it in version control for long periods
+/// (other than in tests etc.)
```
Comment (joshtriplett) [inline library/std/src/macros.rs:189]:
```suggestion
/// (other than in tests and similar).
```
Minor nit.

**Comment by joshtriplett** [comment]
Looks good to me; I particularly appreciate the phrasing of "Debug output from production code is better done with ...", as that makes the intent of `dbg!` quite clear.

**Comment by ijackson** [comment]
I wonder if this will do anything:

@bors r? joshtriplett

**Comment by m-ou-se** [comment]
> I wonder if this will do anything:
> 
> @bors r? joshtriplett

@ijackson FYI: `r?` is not handled by bors, but by rust-highfive. It reacts without being tagged, but does require the `@` for the username. So, `r? @joshtriplett` would have worked.

**Pull request merged**